### DISCUSSION TITLE
Audio and recording

### DISCUSSION THREAD

**Discussion by mbruski** [AlexandreRouma/SDRPlusPlus#978] [Q&A]
Don't know if I missed something but I'm trying to use audio from sdrpp with DireWolf.  Direwolf expects audio to be sampled at 44100Hz.   So I set that value in the sink and DireWolf is happy with the live audio but when I record that audio and play it back with the 'atest' utility, it reports the audio is sampled at 48000Hz and doesn't decode anything even though it decoded the same audio when it was live.   Is there a way to get the recorder to use the sample rate provided in the sink?

Thanks,
Mike

**Comment by AlexandreRouma** [comment]
The recorder records audio at whatever samplerate the sink is using. Change the samplerate in the sink menu if you need something else than the default 48k.

Additionally, perhaps the developers of Direwolf should realize that expecting every audio source to run at one samplerate is silly. Resampling exists for a reason.

**Reply by mbruski** [reply]
I should have noted I'm using sdrpp v1.1.0 built on Ubuntu 20.04 LTS.
I went back and played with it for a good while and no matter what I set the sample rate to in the sink, the resulting audio came out 48000Hz.  There is a signal there but it sounds weaker than compared to the live audio.
I also tried resampling with sox but that just made it worse and I still couldn't get anything to decode.  Sox reported that resampling and dithering caused some 170K samples to be clipped.  I tried recording some other short audio segments with volume reduced and recorder VU level reduced so it's not in the red and still get these warnings during resampling.  Maybe I need to give up the noise reduction features and go back to v1.0.4?  I would be disappointed to switch it all back only to discover the problem still exists with the official release version.
![sdrpp-screen](https://user-images.githubusercontent.com/70450969/215354596-b01d3e1a-568a-4d56-adf5-fc6c3ccd22e1.jpg)
![soxi-screen](https://user-images.githubusercontent.com/70450969/215354604-2540ae47-990e-4f22-b6d2-8ec57819076d.jpg)

**Reply by AlexandreRouma** [reply]
you're not running the latest nightly, a bug like that was fixed earlier this month after the recorder was rewritten.

**Reply by mbruski** [reply]
That might well be.  I pulled my copy on Jan 4.  I'll pull it again and try with the latest.

Thank you.

**Reply by mbruski** [reply]
Just to confirm, that did fix the issue I was seeing.  The recorder is now following the sample rate set in the sink.  Thank you again.

Mike
